test/libsolidity/SemanticTest.cpp

```diff
@@ -383,7 +384,10 @@ TestCase::TestResult SemanticTest::runTest(
 			soltestAssert(
 				deploy(test.call().signature, 0, {}, libraries) && m_transactionSuccessful,
 				"Failed to deploy library " + test.call().signature);
-			libraries[test.call().signature] = m_contractAddress;
+			if (test.call().signature.find(':') == string::npos)
```

Comment:
I'd add a comment to explain why we're doing it.

```suggestion
			# For convenience, in semantic tests we assume that an unqualified name like `L` is equivalent to one
			# with an empty source unit name (`:L`). This is fine because the compiler never uses unqualified
			# names in the Yul code it produces and does no allow `linkersymbol()` at all in inline assembly.
			if (test.call().signature.find(':') == string::npos)
```

Reply:
done